utils/build-script-impl

```diff
@@ -922,138 +1121,156 @@ if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
+                test_this_target=$(not ${SKIP_TEST_IOS_SIMULATOR})
+                ;;
+            appletvos-*)
+                swift_sdk="TVOS"
+                build_for_this_target=$(not ${SKIP_BUILD_TVOS_DEVICE})
+                if [[ ! "${SKIP_TEST_TVOS_HOST}" ]] ; then
+                    test_host_only=1
+                else
+                    test_this_target=
+                fi
+                build_benchmark_this_target=$(not ${SKIP_BUILD_TVOS_DEVICE})
+                ;;
+            appletvsimulator-*)
+                swift_sdk="TVOS_SIMULATOR"
+                build_for_this_target=$(not ${SKIP_BUILD_TVOS_SIMULATOR})
+                test_this_target=$(not ${SKIP_TEST_TVOS_SIMULATOR})
+                ;;
+            watchos-*)
+                swift_sdk="WATCHOS"
+                build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
+                if [[ ! "${SKIP_TEST_WATCHOS_HOST}" ]] ; then
+                    test_host_only=1
+                else
+                    test_this_target=
+                fi
+                build_benchmark_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
+                ;;
+            watchsimulator-*)
+                swift_sdk="WATCHOS_SIMULATOR"
+                build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_SIMULATOR})
+                test_this_target=$(not ${SKIP_TEST_WATCHOS_SIMULATOR})
+                ;;
+            android-*)
+                swift_sdk="ANDROID"
+                build_for_this_target=$(not ${SKIP_BUILD_ANDROID})
+                # FIXME: Allow Android host tests to be enabled/disabled by the
+                #        build script.
                 test_this_target=
-            fi
-            build_benchmark_this_target=$(not ${SKIP_BUILD_WATCHOS_DEVICE})
-            ;;
-        watchsimulator-*)
-            build_for_this_target=$(not ${SKIP_BUILD_WATCHOS_SIMULATOR})
-            test_this_target=$(not ${SKIP_TEST_WATCHOS_SIMULATOR})
-            ;;
-        android-*)
-            build_for_this_target=$(not ${SKIP_BUILD_ANDROID})
-            # FIXME: Allow Android host tests to be enabled/disabled by the
-            #        build script.
-            test_this_target=
-            ;;
-        *)
-            echo "Unknown compiler deployment target: ${deployment_target}"
-            exit 1
-            ;;
-    esac
-    if [[ "${build_for_this_target}" ]] ; then
-        if [[ "${BUILD_SWIFT_STDLIB_UNITTEST_EXTRA}" == "1" ]] ; then
-            SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
-        else
-            if [[ "${VALIDATION_TEST}" == "1" || "${LONG_TEST}" == "1" ]] ; then
-                SWIFT_STDLIB_TARGETS+=("swift-stdlib-${deployment_target}")
+                ;;
+            *)
+                echo "Unknown compiler deployment target: ${stdlib_deployment_target}"
+                exit 1
+                ;;
+        esac
+
+        SWIFT_SDKS+=("${swift_sdk}")
```

Comment:
Yes it will configure for all SDKs because that's what is given in `stdlib-deployment-targets` (see `utils/swift-build-supports/targets.py`). However, it will only build for `macosx-x86_64`.

As I've implemented it now, `stdlib-deployment-targets` controls which targets get configured and `build-stdlib-deployment-targets` controls which get built (default is `all`, meaning all configured, although this is in conjunction with the various `skip-ios`..etc flags as we have them today -- it's an additional filter).

This means you can add the `--ios` flag, for example, and if you have any in your build list (or you didn't restrict the build list) it will build the iOS slices without needing to be reconfigured.